src/Modules/Games/Blef.Modules.Games.Application/Commands/Handlers/CheckHandler.cs

```diff
@@ -6,14 +6,22 @@ namespace Blef.Modules.Games.Application.Commands.Handlers;
 internal sealed class CheckHandler : ICommandHandler<Check>
 {
     private readonly IGamesRepository _games;
+    private readonly ICommandDispatcher _commandDispatcher;
 
-    public CheckHandler(IGamesRepository games) =>
+    public CheckHandler(IGamesRepository games, ICommandDispatcher commandDispatcher)
+    {
         _games = games;
+        _commandDispatcher = commandDispatcher;
+    }
 
     public Task Handle(Check command, CancellationToken cancellation)
     {
         var game = _games.Get(command.GameId);
         game.Check(command.PlayerId);
+
+        // if finished then start new one.
```

Comment:
I wonder that if finished then the new one game shouldn't be started explicite by client (not implicite like that).